lib/server.ts

```diff
@@ -223,8 +240,51 @@ export function startServer(server, opts) {
 `);
     log.success(`bos-workspace running on port ${opts.port}!`);
   })
-    .on("error", (err: Error) => {
-      log.error(err.message);
-      process.exit(1);
+    .on("error", async (err: any) => {
+      if (err.code === "EADDRINUSE") {
+        log.warn(err.message);
+
+        const devJson = await readJson(devJsonPath, { throws: false });
+
+        const postData = JSON.stringify(devJson);
+        const options = {
+          hostname: '127.0.0.1',
+          port: opts.port,
+          path: `/api/loader`,
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': Buffer.byteLength(postData),
+          },
+        };
+
+        const req = http.request(options, (res) => {
```

Comment:
I don't think you need to make a request to the server again; you're already in the server.

We want to make sure this new repository we're in gets added to the dev + build commands, we still want `dev` or `devMutli` to run and watch for changes with the new bos.workspace.json or bos.config.json we're introducing. 

You've gotten to this part because of running the dev command -- what if we caught the server sooner, within the dev command itself? And used absolute paths

https://github.com/NEARBuilders/bos-workspace/blob/5380ef82f51b1f9595fc4c9a4c5fdf8c69390e11/lib/dev.ts#L31

Reply:
We essentially wanna get this new project added to the list here:

https://github.com/NEARBuilders/bos-workspace/blob/5380ef82f51b1f9595fc4c9a4c5fdf8c69390e11/lib/dev.ts#L64